.github/workflows/test.yml

```
@@ -6,10 +6,36 @@ on:
     types: [opened, synchronize]
     branches-ignore: [staging, production]
 
+env:
+  # Number of parallel jobs for jest tests
+  CHUNKS: 3
 jobs:
+  config:
+    runs-on: ubuntu-latest
+    name: Define matrix parameters
+    outputs:
+      MATRIX: ${{ steps.set-matrix.outputs.MATRIX }}
+    steps:
+      - name: Set Matrix
+        id: set-matrix
+        uses: actions/github-script@v6
+        with:
+          # Generate matrix array i.e. [0, 1, 2, ...., CHUNKS - 1] for test job
+          script: |
+            core.setOutput('MATRIX', Array.from({ length: Number(process.env.CHUNKS) }, (v, i) => i));
+
   test:
+    needs: config
     if: ${{ github.actor != 'OSBotify' || github.event_name == 'workflow_call' }}
     runs-on: ubuntu-latest
+    name: test (job ${{ fromJSON(matrix.chunk) + 1 }})
```

Error is being thrown here. I think it should be
```suggestion
    name: test (job ${{ Number(matrix.chunk) + 1 }})
```

---

@parasharrajat  I tried `Number()` locally and ran github action validate command and it is failing again. Linting has some parsing issue. Also, I don't think `Number` will work in github action expression. It's not metioned in available functions for expressions. 

I tried this in my own test repository and it works. IMHO, it's issue with Linting tool.

---

Linting has issues with `+` sign

workflows/test.yml:31:46: got unexpected character '+' while lexing expression, expecting 'a'..'z', 'A'..'Z', '_', '0'..'9', ''', '}', '(', ')', '[', ']', '.', '!', '<', '>', '=', '&', '|', '*', ',', ' ' [expression]

---

Other option is to create another variable like `JEST_CHUNKS` or something that you suggest and we can calculate and set in github js script where we are setting `MATRIX` environment variable for matrix config.
```
script: |
            core.setOutput('MATRIX', Array.from({ length: Number(process.env.CHUNKS) }, (v, i) => i));
            core.setOutput('JEST_CHUNKS', Number(process.env.CHUNKS) - 1);



- name: All Unit Tests
        if: ${{ fromJSON(matrix.chunk) < fromJSON(env.JEST_CHUNKS) }}
        run: npx jest --listTests --json | jq -cM '[_nwise(length / ${{ fromJSON(env.JEST_CHUNKS) }} | ceil)]' | jq '.[${{ fromJSON(matrix.chunk) }}] | .[] | @text' | xargs npm test

      - name: Pull Request Tests
        if: ${{ fromJSON(matrix.chunk) == fromJSON(env.JEST_CHUNKS) }}
        run: tests/unit/getPullRequestsMergedBetweenTest.sh
```

---

and even if we do the above, we can't start job name from 1 i.e. `job 1` because we again have to add 1 to make it right and using `+` is not allowed in linting rule. Tried different methods locally but did not work.

---

Let me check this

---

Ok, `+` is not a valid operator in Github actions. 

May be we can start with this Matrix and then job name will be correct and adjust the other operations to meet this. 
```yaml
core.setOutput('MATRIX', Array.from({ length: Number(process.env.CHUNKS) }, (v, i) => i  +1));
```

If this is not working, then `JEST_CHUNKS` approach is fine too.

---

Here is output for above command

```
[
  [],
  [  
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/loginTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/ReportUtilsTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/fetchFreePlanVerifiedBankAccountTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/createOrUpdateStagingDeployTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/checkDeployBlockersTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/awaitStagingDeploysTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/MigrationTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/NetworkTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/RequestTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/WalletTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/OptionsListUtilsTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/actions/SessionTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/actions/ReportTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/EmojiRegexTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/isStagingDeployLockedTest.js"
  ],
  [
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/isPullRequestMergeableTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/TranslateTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/GithubUtilsTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/actions/ReimbursementAccountTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/PaymentUtilsTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/nativeVersionUpdaterTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/enhanceParametersTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/versionUpdaterTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/GooglePlacesUtilsTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/CurrencySymbolUtilsTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/NumberUtilsTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/LocalePhoneNumberTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/GitUtilsTest.js",
    "/mnt/c/Users/BIG BANG/OneDrive/Desktop/Upwork/App/tests/unit/CardUtilsTest.js"
  ]
]
```

---

IMO, we should increase CHUNKS to 4 at-least or 5 otherwise there will be negligible gain in execution time because with CHUNKS: 3, jest test will be divided into two jobs.

---

So there will be a total of 3 jobs. One is for Pull request tests. If this is taking less than 5 minutes in total then I guess two is fine too. let's see when checks are triggered.